### ISSUE TITLE
QA from RC2

### ISSUE THREAD

**Issue by cagulas** [OpenSRP/opensrp-client-uganda-hpv#35]
- [x] MB: Add the "undo" option back to vaccinations
- [x] RW: On a tablet, the app crashes when you tap on a phone number on the profile view. This could be because the tablet is wifi only. ~~CG: Ideally, a toaster message would appear saying something like "Phone app not available".~~ RW: For tablets without phones/calling, we can open up a "copy to clipboard dialog (as seen below on long press on my tablet's Contacts app). Alternatively, in a nice to have situation, if the tablet is able to detect whether the number can be used in WhatsApp or other messaging app, it can additionally have options to select those apps to use the number in.
  ![copytoclipboard](https://user-images.githubusercontent.com/1584163/40428844-383d413c-5e6f-11e8-90cf-3a0f367afd3f.png)
- [x] CG: I'm noticing that the HPV app pops up sometimes when I unlock my phone to use it, even though the app was not left open when I put my phone down. Is this an issue, or should I take this as a message from the beyond?
- [x] RW: List screen suggestions (search bar, location carrot icon positioning, due button label styling)
  ![listedits](https://user-images.githubusercontent.com/1584163/40430365-fec7e296-5e72-11e8-9afa-4b6762f36a9e.png)
- [x] RW: Contact screen suggestions (add contact label, dose info spacing)
  ![editcontacts](https://user-images.githubusercontent.com/1584163/40430366-ff00c5e8-5e72-11e8-8bf6-94a23cd9029b.png)
- [x] RW: Search behavior: if I tap the enter key on the keyboard after typing something in, right now it goes to a "no results" view, but it really should just keep the same search parameters and hide the keyboard.. the same behavior of hitting enter after search if the search bar wasn't filtering as you type.
  ![screenshot_20180523-151347](https://user-images.githubusercontent.com/1584163/40430396-0f87fc10-5e73-11e8-9330-1229de15d0bd.png)
- [x] RW: The type size on the service location dialog seems... big. Bigger than any other type in the app. Probably should make that about same as the name on the girl detail, and the dialog wider (almost full screen width like the copy to clipboard screen above) so long names can fit.
  ![screenshot_20180523-151925](https://user-images.githubusercontent.com/1584163/40430428-204ccfd0-5e73-11e8-86c6-45c69f78e380.png)
- [x] RW: Could we make the placeholder grey box for the avatar photo clickable to take a photo and also add the camera icon in the center, and change the colors to match the placholder avatar (#5b9ab6 on #b7d3df). Also the type is big here but I'm okay with that since then it can be shown to the girl for spelling verification if that ever makes sense to do.
  ![screenshot_20180523-151618](https://user-images.githubusercontent.com/1584163/40430467-2d6a4b3e-5e73-11e8-922c-4dc7a00ccd76.png)
- [X] ~~RW: can we more easily make the phone input be smarter or is it not worthwhile doing for this version? Or was that a client requirement? It would be nice to have the number input be smart enough to translate whatever format the person enters for example, if they use “256” plus 9 numbers, it should be a valid entry vs. requiring “0”.~~ Removed as requirement
- [x] RW: is "Manual sync" good or just "Sync" okay for the wording in the top right dropdown?

**Comment by ndegwamartin** [comment]
@rowo for phone number update concerning the 256, I think most people when they enter phone numbers in their locale usually just leave out the country code prefix . Also, most functionality to do with the native forms is highly coupled with the Library we use so any modifications imply its update. 

However it can still be done. we would have to update the library and perhaps create a custom validator for this. This way we won't affect other projects using the library.

(Most)Other updates though will reflect on the other projects .

**Comment by ndegwamartin** [comment]
@rowo also i've change the manual sync to read just 'Sync' and match zeir

**Comment by cagulas** [comment]
I don't think the 256 phone number feature is important to do. @rowo what do you think?

**Comment by ndegwamartin** [comment]
@cagulas the app pops up due to the sync implementation whose intent is to sync team data even when you are not using the app. we've updated it to be a bit more efficient but you probably noticed it because the debug app syncs every 2 minutes.

**Comment by cagulas** [comment]
@ndegwamartin why doesn't this also happen with the ZEIR app then?

**Comment by ndegwamartin** [comment]
Good question @cagulas . There are the aforementioned updates we made to the function that were the difference with zeir , we can test see if the issue recurs in the next release candidate.

**Comment by rowo** [comment]
@ndegwamartin @cagulas right, yeah anything where I'm asking a question in my comment is not a final decision or anything. Definitely depends on your opinions and feasibility.

**Comment by mberg** [comment]
Just to clarify.  I think we should support undo but we can do that in
future apps.  We don't need to correct for HPV to get this out.  We can do
that once we have a better sense of when it will be deployed in the field.

**Comment by cagulas** [comment]
@ndegwamartin Do you want to send me the apk tomorrow with everything you've done so far in this issue? I reckon this will be good to send off to the client.

**Comment by ndegwamartin** [comment]
actually the undo vaccine function is almost done @vincent-karuri has been working on that. we should have an RC3 apk ready today with the checked items on this issue's checklist included.

**Comment by vincent-karuri** [comment]
the undo functionality button design and positioning still needs to be determined, waiting on @rowo

**Comment by vincent-karuri** [comment]
currently it looks like this, maybe not satisfactory
<img width="411" alt="screen shot 2018-05-24 at 18 05 55" src="https://user-images.githubusercontent.com/33488286/40536518-1bb689ae-6015-11e8-9d2c-1548d39cca56.png">

**Comment by rowo** [comment]
Left aligned

![screen-shot-2018-05-24-at-18 05 55](https://user-images.githubusercontent.com/1584163/40551434-e60d64c6-600a-11e8-8b73-e0d97f13083c.png)

Centered

![screen-shot-2018-05-24-at-18 05 55](https://user-images.githubusercontent.com/1584163/40551455-f899493e-600a-11e8-9fd5-e6257de0125b.png)

**Comment by cagulas** [comment]
@rowo Both Martin and I are not able to replicate the first one about hitting enter. It might be due to differences in the keyboard on different devices. If it's ok with you, we'll leave that one in the interest of getting this out today and wrapped up. The second one can be addressed easily.

**Comment by ndegwamartin** [comment]
@rowo for the search issue check if you might be having any special settings on your device, for instance on the samsung tablet we have an Auto Replace setting that seems to override any application's Enter key logic

**Comment by cagulas** [comment]
Lgtm
